src/producer.rs

```diff
@@ -127,7 +128,7 @@ pub struct ProducerOptions {
     /// batch message size
     pub batch_size: Option<u32>,
     /// algorithm used to compress the messages
-    pub compression: Option<proto::CompressionType>,
+    pub compression: Option<Compression>,
```

Comment:
This looks like a breaking change, is it possible to add a new field and deprecate this one instead of changing its type directly?

Reply:
With this comment, I tend to revert 34bf64ae7edb318b9651d89698b9587ffab2f7bb which applies @FlorentinDUBOIS's comment.

For breaking change, I make it with these trade-off considerations:

1. `compression` is the best name for this config, if we keep b/w compatibility, we need to choose a new name like `compression_mode` or `compression_strategy`.
2. This library - although the latest release is 4.2.0 - is still incomplete for production and I think exposing `proto::*` is a temporary solution.
3. If - with 34bf64ae7edb318b9651d89698b9587ffab2f7bb reverted - we use a wrapper struct to this field, and as long as we implement `Default` for it, it can be b/w compatible.
4. Deprecating and fallback logics are also overhead/debt.

Reply:
Fixed in https://github.com/streamnative/pulsar-rs/pull/240/commits/c889a08fd9aa34cf6aab74c3e1d09c67bca77b1e

Reply:
I see. Maybe we can implement the `From<proto::CompressionType >` trait for `Compression` to make the code migration easier.

Reply:
Good point. Let me try it out.

Reply:
@fantapsody how can you imagine it eases the case?

Perhaps `compression: Some(proto::CompressionType::Snappy.into())`? Since it also touches the code anyway, I think it's not quite better than `compression: Some(Compression::Snappy(CompressionSnappy::default()))` with `use compression::...`.

The problem here is, `proto::CompressionType` is available always, and if we support such conversion, we need to play some runtime check which we removed at https://github.com/streamnative/pulsar-rs/pull/240#discussion_r1027263857 again. I don't think it's a valuable trade-off.